packages/ast-spec/src/parameter/ArrayPattern/spec.ts

```diff
@@ -7,7 +7,7 @@ import type { DestructuringPattern } from '../../unions/DestructuringPattern';
 export interface ArrayPattern extends BaseNode {
   type: AST_NODE_TYPES.ArrayPattern;
   elements: (DestructuringPattern | null)[];
-  typeAnnotation?: TSTypeAnnotation;
-  optional?: boolean;
-  decorators?: Decorator[];
+  typeAnnotation: TSTypeAnnotation | undefined;
+  optional: boolean;
+  decorators: Decorator[];
```

Comment:
Just because I was looking - in TS 5.0 decorators on function parameters are a semantic error under the new decorators parsing

[playground](https://www.typescriptlang.org/play?experimentalDecorators=false&emitDecoratorMetadata=false&ts=5.0.0-dev.20230131#code/CYUwxgNghgTiAEYD2A7AzgF3qZAueUKAngNwBQZAZgK4pgYCWq8lSSAFGfPAAI5LwA2lAC6+QZhgMUAcxEAaLr37wARvAC88AIyLufcAIDeYAL74T+SdJmm9yw-AB0L4IoCU8I-FMVIUNDR4AGEvJWR0DBhqeiQYTm59FWExIWtZBSUkx3UtXSyHZC8zCzArKJs7AoMigAcpADcoDARgcqlZe2yilycQe09vX18yIA)

But until `experimentalDecorators` is removed, we'll need to keep this.

Reply:
> until `experimentalDecorators` is removed

Heh. Maybe in TS 50.0.